### PR TITLE
DOCS-12254-dw-pluck-function-example-kt

### PR DESCRIPTION
**Pull request by kevintroller**
DW pluck function example

### CODE REVIEW

File: modules/ROOT/pages/dw-core-functions-pluck.adoc [hunk 1]
```diff
@@ -92,9 +92,58 @@ var readXml = read("<prices>
 }
 ----
 
+=== Example
+
+This example uses the `pluck` function to iterate over the key-value pairs in the source and outputting the values of those pairs into an array of `Test` objects.
```
Comment (dukesphere) [inline modules/ROOT/pages/dw-core-functions-pluck.adoc:97]:
```suggestion
This example uses the `pluck` function to iterate over the key-value pairs in the source and outputs the values of those pairs into an array of `Test` objects.
```

**Review by teofr — DISMISSED** [review]

File: modules/ROOT/pages/dataweave-cookbook-pluck.adoc [hunk 2]
```diff
@@ -0,0 +1,132 @@
+= Extract Key/Value Pairs with Pluck Function
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: studio, anypoint, transform, transformer, format, dataweave, data weave, output structure, mime type, mime
+
+The following Mule app example shows how to use the DataWeave `pluck` function to extract key/value pairs from a JSON payload and store them into flow variables with variable name as the keys and their corresponding values.
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook-pluck.adoc:7]:
```suggestion
The following Mule app example shows how to use the DataWeave `pluck` function to extract key-value pairs from a JSON payload and store them into Mule event variables with variable names as the keys and their corresponding values.
```

Comment (dukesphere) [reply modules/ROOT/pages/dataweave-cookbook-pluck.adoc:7]: We don't use the term "flow variable" in Mule 4 as far as I know. Could also be "store into a Mule variable" or something like that.

File: modules/ROOT/pages/dataweave-cookbook-pluck.adoc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+The Mule app consists of:
+
+* A Scheduler component that triggers the flow
+* A Set Payload component that sets the JSON payload
+* A Transform Message component that transforms the content read from the Set Payload component by extracting the key/value pairs with the `pluck` function and put them into an array
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook-pluck.adoc:13]:
```suggestion
* A Transform Message component that transforms the content read from the Set Payload component by extracting the key-value pairs with the `pluck` function and putting them into an array
```

File: modules/ROOT/pages/dataweave-cookbook-pluck.adoc [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+* A For Each scope component that iterates the array payload content
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook-pluck.adoc:14]:
```suggestion
* A For Each scope component that iterates over the array
```

File: modules/ROOT/pages/dataweave-cookbook-pluck.adoc [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+* A Set Variable component that sets the flow variable name as the Key and variable value as the Value in the JSON array
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook-pluck.adoc:15]:
```suggestion
* A Set Variable component that sets the variable name as the Key and variable value as the Value in the JSON array
```

File: modules/ROOT/pages/dataweave-cookbook-pluck.adoc [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  {
+    "Test": {
+      "Key": "dept",
+      "Value": "support"
+    }
+  }
+]
+----
+
+Then, the For Each scope component iterates over the transformed values and the app logs the values of the variable results:
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook-pluck.adoc:108]:
```suggestion
The For Each scope component iterates over the transformed values, and the app logs the values of the variable results, as follows:
```

File: modules/ROOT/pages/dataweave-cookbook.adoc [hunk 7]
```diff
@@ -169,6 +169,9 @@ These examples show how to use DataWeave to extract and transform data in MuleSo
 //read-connector
 | xref:dataweave-cookbook-writer-prop-mule.adoc[Use a Mule Variable as a Configuration Value]
 | Use a Mule variable as a configuration value.
+//pluck-function
+| xref:dataweave-cookbook-pluck.adoc[Extract Key/Value Pairs with Pluck Function]
+| Use the DataWeave `pluck` function to extract key/value pairs from a JSON payload and store them into flow variables.
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook.adoc:174]:
```suggestion
| Use the DataWeave `pluck` function to extract key-value pairs from a JSON payload and store them into Mule event variables.
```

File: modules/ROOT/pages/dataweave-cookbook-pluck.adoc [hunk 8]
```diff
@@ -0,0 +1,132 @@
+= Extract Key/Value Pairs with Pluck Function
```
Comment (dukesphere) [inline modules/ROOT/pages/dataweave-cookbook-pluck.adoc:1]:
```suggestion
= Extract Key-Value Pairs with the `pluck` Function
```

**Review by dukesphere — COMMENTED** [review]
some suggestions

**Review by dukesphere — DISMISSED** [review]

**Review by dukesphere — APPROVED** [review]

**Comment by kevintroller** [comment]
Closing this PR because the Traceability check did not run at all, despite updating and refreshing the URL many times. Created new PR https://github.com/mulesoft/docs-dataweave/pull/83

**Pull request closed without merging**